app/assets/javascripts/components/CreateCase.jsx

```diff
@@ -0,0 +1,213 @@
+class CreateCase extends React.Component {
+    constructor(props) {
+        super();
+        this._handleChange = this._handleChange.bind(this);
+        this._handleSubmit = this._handleSubmit.bind(this);
+        this.state = {
+            showForm: true,
+            message: '',
+            error: ''
+        };
+    }
+
+    _handleChange = (e) => {
+        this.setState({ [$(e.target).attr("name")] : $(e.target).val() });
+    }
+
+    _handleSubmit = (e) => {
+        e.preventDefault();
+        const payload = {
+            "user_id": this.props.user.id,
+            "type_of_case": this.state.type_of_case,
+            "pro_bono_placement": this.state.pro_bono_placement,
+            "grant": this.state.grant,
+            "initial_invoice_date": this.state.initial_invoice_date,
+            "last_invoice_date": this.state.last_invoice_date,
+            "date_rec_initial_disbursement": this.state.date_rec_initial_disbursement,
+            "date_rec_last_disbursement": this.state.date_rec_last_disbursement,
+            "case_tracking": this.state.case_tracking,
+            "program": this.state.program,
+            "legal_case_name": this.state.legal_case_name,
+            "judge_assigned": this.state.judge_assigned,
+            "trial_attorney": this.state.trial_attorney,
+            "case_progress": this.state.case_progress,
+            "date_biometrics_done": this.state.date_biometrics_done,
+            "lodge_or_rn_date": this.state.lodge_or_rn_date,
+            "date_mta_filed": this.state.date_mta_filed,
+            "asylum_officer": this.state.asylum_officer,
+            "nexus_granted": this.state.nexus_granted,
+            "nexus_argued": this.state.nexus_argued,
+            "case_outcome": this.state.case_outcome,
+            "case_outcome_achieved": this.state.case_outcome_achieved,
+            "date_of_outcome": this.state.date_of_outcome
+        };
+
+        Requester.post('http://localhost:3000/api/cases', payload).then((data) => {
```

Comment:
You should have all the routes in a file called `routes.js` or similar and then just import them from there so you don't have multiple URL strings in different places. If you decide to change your backend later then you'd have to go around and fix all the instances of the URL strings while if you collect all of them in one class you can fix everything there.

Reply:
since we don't have a `routes.js` file yet it'd be enough to just drop the `localhost:3000` portion and just have `/api/cases`